main.tf

```diff
@@ -437,7 +437,11 @@ resource "aws_security_group_rule" "ingress_with_prefix_list_ids" {
   security_group_id = local.this_sg_id
   type              = "ingress"
 
-  prefix_list_ids = var.ingress_prefix_list_ids
+  prefix_list_ids = coalescelist(
+    var.ingress_with_prefix_list_ids[count.index]["prefix_list_ids"],
+    var.ingress_prefix_list_ids
+  )
```

Comment:
`coalescelist` can trigger an error if both lists are empty.
Maybe it's a better idea to use `lookup` with the value of `var.ingress_prefix_list_ids` as default value.